site/cypress/e2e/Cypress-Gradeable/late_submission_warning_messages.spec.js

```diff
@@ -0,0 +1,366 @@
+            .type(getCurrentTime())
+            .type('{enter}');
+        cy.get('#save_status', {timeout:20000}).should('have.text', 'All Changes Saved');
+        cy.logout();
+        SubmitAndCheckMessage('non_team','upload_file2','invalid_1_day_late');
+        cy.login('student');
+        cy.visit(['sample', 'gradeable', gradeable]);
+        cy.get('#do_not_grade').should('exist').click();
+    });
+
+    it('should show a confirmation message for the first submission', () => {
+        //Part 1/2 of a test case
+        //The first submission will be done 3 days after the due date and use 2 valid late days
+        cy.login('instructor');
+        giveExtentions(gradeable);
+        giveLateDays(getCurrentTime('threeDaysAgo'),'student'); //Give valid late days (the current ones are after the original due date)
+        cy.visit(['sample', 'gradeable',gradeable,'update?nav_tab=5']);
+        cy.get('#late_days')
+            .clear()
+            .type(3)
+            .type('{enter}');
+        cy.get('#save_status', {timeout:20000}).should('have.text', 'All Changes Saved');
+        cy.get('#date_due')
+            .should('exist')
+            .clear()
+            .type(getCurrentTime('threeDaysAgo'))
+            .type('{enter}');
+        cy.get('#save_status', {timeout:20000}).should('have.text', 'All Changes Saved');
+        cy.logout();
+        SubmitAndCheckMessage('non_team','upload_file1','valid_usage','2_days_late+extention');
+    });
+    it('should show a warning message for the second submission ', () => {
+        /*Part 2/2 of a test case
+        This submission is invalid because the late days remaining are earned at the extention date,
+        not the original due date.*/
+        cy.login('student');
+        cy.wait(5000);//to avoid flakyness and to make sure 4 days have passed (since in getcurrenttime(), there is an extra 5 seconds)
+        SubmitAndCheckMessage('non_team','upload_file2','invalid_4_days_late');
+        cy.login('instructor');
+        cy.visit(['sample', 'gradeable',gradeable,'update?nav_tab=5']);
+        cy.get('#no_late_submission').should('exist').click(); //disable late submissions
+        cy.get('#save_status', {timeout:20000}).should('have.text', 'All Changes Saved');
+    });
+});
+
+describe('Test warning messages for team gradeable', () => {
+
+    it('should create team gradeable for testing', () => {
+        cy.login('instructor');
+        cy.visit(['sample', 'gradeable']);
+        // Enter gradeable info
+        cy.get('#g_title').type(team_gradeable);
+        cy.get('#g_id').type(team_gradeable);
+        cy.get('#radio_ef_student_upload').check().click();
+        cy.get('#team_yes_radio',{timeout:20000}).should('exist').check().click();
+        // Create Gradeable
+        cy.get('#create-gradeable-btn').click();
+
+        //Date page
+        cy.get('#page_5_nav').should('exist').click();
+        cy.get('#date_ta_view')
+            .should('exist')
```

Comment:
I believe in places like this, asserting existence is not needed, as the test will fail if it doesn't exist already